### PR TITLE
driver: Add the MCP320x ADC driver

### PR DESCRIPTION
**Pull request by DipSwitch**
Add a device driver for the Microchip MCP320x SPI ADC.

### CODE REVIEW

File: drivers/mcp320x/mcp320x.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define SPI_SPEED                  SPI_SPEED_5MHZ
+
+/**
+ * @brief enum representation of the available modes
+ */
+typedef enum {
+    Single = 0x2,
```
Comment (thomaseichinger) [inline drivers/mcp320x/mcp320x.c]: Please un-capitalise.

**Comment by thomaseichinger** [comment]
@DipSwitch I'm very sorry for the long delay. The code looks good, only one minor comment. Please also add a test application in `tests/`. Would you mind to also adapt #4271 and give it a try?

**Comment by DipSwitch** [comment]
I'll write a test somewhere later this week. And try implement an `mcp320x_param.h`

**Comment by OlegHahm** [comment]
@DipSwitch, any luck?

**Comment by OlegHahm** [comment]
What about this one? Chances to get this ready for next week?

**Comment by DipSwitch** [comment]
Nope, don't have the hardware atm :)

**Comment by kYc0o** [comment]
Feature freeze -> postponed.

**Comment by miri64** [comment]
Feature freeze -> postponed.

**Comment by OlegHahm** [comment]
Any news here?

**Comment by smlng** [comment]
trigger Murdock, but also remove release milestone as no reaction/progress for >  a year

**Comment by miri64** [comment]
No significant activity for over a year. Let's close this as memo.

**Pull request closed without merging**